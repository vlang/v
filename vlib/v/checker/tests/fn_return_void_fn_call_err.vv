fn main() {
	install_command(['github:some/thing'], 1) or { panic(err) }
}

pub fn install_command(args []string, verbosity int) ! {
	if args.len == 0 {
		return error('${@FN} requires an argument')
	}
	component := args[0]
	if component.count(':') == 0 {
		return install_command(['github:${component}'], verbosity)
	}
	source := component.all_before(':')
	if source != 'github' {
		return error('${@FN} unknown source `${source}`')
	}
	unit := component.all_after(':')
	return match source {
		'github' {
			install_from_github(unit, verbosity)!
		}
		else {
			error('${@FN} unknown source `${source}`')
		}
	}
}

fn install_from_github(unit string, verbosity int) ! {
	dump(unit)
}
