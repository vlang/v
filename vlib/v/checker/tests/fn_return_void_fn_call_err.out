vlib/v/checker/tests/fn_return_void_fn_call_err.vv:20:4: error: `install_from_github(unit, verbosity)!` used as value
   18 |     return match source {
   19 |         'github' {
   20 |             install_from_github(unit, verbosity)!
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   21 |         }
   22 |         else {
